src/state/crdt/documents/operations.ts

```diff
@@ -407,7 +407,18 @@ export function generateMoveDown(
 }
 
 // helper type to allow us to omit properties from a union type
-type DistributiveOmit<T, K extends keyof T> = T extends T ? Omit<T, K> : never;
+export type DistributiveOmit<
+  T,
+  K extends T extends unknown ? keyof T : never,
+> = T extends unknown
+  ? {
+      [P in keyof Omit<T, Extract<keyof T, K>>]: Omit<
+        T,
+        Extract<keyof T, K>
+      >[P];
+    }
+  : never;
+
```

Comment:
Actually with the `kind` prop for the image this can eventually be rolled back.

The problem was that you can only omit props that were there for all types.
Not it would just drop the prop across all of the union types no matter if the prop does not exist for one of them.

Reply:
`kind` is gone again. So we need it.

If we roll it back `UpdateElementPatch` will break.